### PR TITLE
Bids table sf

### PR DESCRIPTION
**Pull request by StephanieFriend**
### Type of change made:
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Styling - no new features
### Why is this change required? What problem does it solve?
Started with styling for Python standards. Created bids table with connection with items table. Moved on to creating full CRUD functionality for bids table. All new functionality is tested.
### Were there any challenges that arose while implementing this feature? If so, how were they addressed?
Had issues with migrating locally. We got together as a team and helped fix those so I can verify everything works before I created this PR

### CODE REVIEW

File: application/__init__.py [hunk 1]
```diff
@@ -1,11 +1,11 @@
-import os
-from flask import Flask, request, jsonify
+from flask import Flask
```
Comment (reid-andrew) [inline application/__init__.py:1]: Good catch cleaning up these import statements which are now not being used in this file.

File: application/controllers/bids.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+        return bid
+
+    @marshal_with(bid_fields)
+    def delete(self, bid_id=None):
+        bid = Bid.query.get(bid_id)
+
+        db.session.delete(bid)
+        db.session.commit()
+
+        return bid
```
Comment (reid-andrew) [inline application/controllers/bids.py:144]: Nice job keeping this structure consistent with how we set up the Items controller.

File: application/models/item.py [hunk 3]
```diff
@@ -13,5 +14,7 @@ class Item(db.Model):
     charity = db.Column(db.String())
     image = db.Column(db.String())
 
+    bids = db.relationship('Bid', backref='items', lazy='select')
+
     def __repr__(self):
         return '<id {}>'.format(self.id)
```
Comment (reid-andrew) [inline application/models/item.py:20]: Thanks for cleaning up the formatting on these.

**Review by reid-andrew — APPROVED** [review]

File: application/controllers/items.py [hunk 4]
```diff
@@ -28,58 +43,59 @@
...
-    help='name parameter is required'
+    help='title parameter is required'
 )
 item_post_parser.add_argument(
     'description',
     type=str,
     required=True,
     location=['json'],
-    help='name parameter is required'
+    help='description parameter is required'
```
Comment (kmcgrevey) [inline application/controllers/items.py:53]: good to get these `helps` cleaned up with their corresponding fields

File: application/models/bid.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    bidder_name = db.Column(db.String())
+    bidder_email = db.Column(db.String())
+    amount = db.Column(db.Float())
+    street_address = db.Column(db.String())
+    city = db.Column(db.String())
+    state = db.Column(db.String())
+    zip_code = db.Column(db.String())
+    receipt = db.Column(db.String())
+
+    item_id = db.Column(
```
Comment (kmcgrevey) [inline application/models/bid.py:17]: Foreign key syntax

File: application/models/bid.py [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+        db.Integer,
+        db.ForeignKey('items.id'),
+        nullable=False
+    )
+
```
Comment (kmcgrevey) [inline application/models/bid.py:22]: This is the foreign key syntax

File: application/models/item.py [hunk 7]
```diff
@@ -13,5 +14,7 @@ class Item(db.Model):
     charity = db.Column(db.String())
     image = db.Column(db.String())
 
+    bids = db.relationship('Bid', backref='items', lazy='select')
+
```
Comment (kmcgrevey) [inline application/models/item.py:18]: corresponding syntax for the `bids` foreign key

File: tests/test_bids.py [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+        bid = Bid(
+            bidder_name="Testy McTesterson",
+            bidder_email="testymctesterson@example.com",
+            amount=300.00,
+            street_address="123 This Is My Street Way",
+            city="My City",
+            state="Colorado",
+            zip_code="12345",
+            receipt="img.ul",
+            item_id=1
```
Comment (kmcgrevey) [inline tests/test_bids.py:38]: Hardcoding this `item_id` in for now and working.  Could be an interpolated Flask syntax similar to Rails to call the `id` of the test-created object?

**Review by kmcgrevey — APPROVED** [review]

**Pull request merged**